src/extension.ts

```diff
@@ -3,34 +3,54 @@
 *  Licensed under the MIT License.
 *--------------------------------------------------------------------------------------------*/
 
-import * as languageclient from 'vscode-languageclient';
-import * as logger from './logger';
 import * as path from 'path';
-import * as schemacontributor from './schema-contributor';
 import * as vscode from 'vscode';
-import * as schemaassociationservice from './schema-association-service';
-import TelemetryReporter from 'vscode-extension-telemetry';
-const fs = require('fs');
+import { createTelemetryReporter, callWithTelemetryAndErrorHandling, IActionContext, AzureUserInput, registerUIExtensionVariables } from 'vscode-azureextensionui';
+import * as languageclient from 'vscode-languageclient';
+
 import { activateConfigurePipeline } from './configure/activate';
+import { extensionVariables } from './configure/model/models';
+import * as logger from './logger';
+import * as schemaassociationservice from './schema-association-service';
+import * as schemacontributor from './schema-contributor';
 
-const myExtensionId = 'azure-pipelines';
-const telemetryVersion = generateVersionString(vscode.extensions.getExtension(`ms-azure-devops.${myExtensionId}`));
-const telemetryKey = 'ae672644-d394-497c-8c57-98f6eac32342';
+let perfStats = {
+    loadStartTime: Date.now(),
+    loadEndTime: undefined
+};
 
-let reporter: TelemetryReporter;
+const configurePipelineEnabled: boolean = vscode.workspace.getConfiguration('[azure-pipelines]', null).get('configure') ? true : false;
 
 export async function activate(context: vscode.ExtensionContext) {
     logger.log('Extension has been activated!', 'ExtensionActivated');
+    setTelemetryReporter(context);
+
+    await callWithTelemetryAndErrorHandling('azurePipelines.activate', async (activateContext: IActionContext) => {
+        activateContext.telemetry.properties['configurePipelineEnabled'] = `${configurePipelineEnabled}`;
+        await activateYmlContributor(context, activateContext);
+        if (configurePipelineEnabled) {
+            await activateConfigurePipeline();
+        }
+    });
 
-    logger.log(`Spinning up telemetry client for id ${myExtensionId}, version ${telemetryVersion}`);
-    reporter = new TelemetryReporter(myExtensionId, telemetryVersion, telemetryKey);
-    context.subscriptions.push(reporter);
+    return schemacontributor.schemaContributor;
+}
+
+function setTelemetryReporter(context: vscode.ExtensionContext) {
+    // Register ui extension variables is required to be done for telemetry to start flowing for extension activation and other events.
+    // It also facilitates registering command and called events telemetry.
+    extensionVariables.reporter = createTelemetryReporter(context);
```

Comment:
except for first line others do not seem to have anything to do with telemetry

Reply:
Removed them and moved to another function